### PR TITLE
Split 90 deg tool, fixing an issue (#142)

### PR DESCRIPTION
**Pull request by romaxa**
out

### CODE REVIEW

**Comment by supersonicclay** [comment]
Would it make more sense for this to be a `modeConfig` option on the existing `spilt` mode? Seems like there would be a good deal of overlap.

**Comment by romaxa** [comment]
Originally I tried to do it as the mode-config option but ran into the issue with nebula example heavily dependent on mode name and how it interacts with menu options. So I decided to not refactor that part, but just extend classes to avoid code duplication.

I can think about adding feature to distinguish tool options as toolName+Mode config parameters if you think we should do it in this CR, otherwise, I'll create another issue for nebula.

File: docs/api-reference/mode-handlers/overview.md [hunk 1]
```diff
@@ -161,6 +161,9 @@ The following options can be provided in the `modeConfig` object:
   * If unit: `x` means the unit used for the spacing would be of `x` unit.
   * Default unit is `centimeters`
 
+* `lock90Degree`
```
Comment (supersonicclay) [inline docs/api-reference/mode-handlers/overview.md]: I like the way deck.gl does [its reference documentation](http://deck.gl/#/documentation/deckgl-api-reference/layers/layer?section=basic-properties). Specify name, type, optional/required, and default. Something like:

```
* `lock90Degree` (Boolean, optional)
  * Default: `false`

If true, all angles will be guaranteed to be 90 degrees.
```

File: examples/deck/example.js [hunk 2]
```diff
@@ -218,6 +229,7 @@ export default class Example extends Component<
               <option value="draw90DegreePolygon">draw90DegreePolygon</option>
               <option value="drawRectangle">drawRectangle</option>
               <option value="split">split</option>
+              <option value="split90Degree">split90Degree</option>
```
Comment (supersonicclay) [inline examples/deck/example.js]: No longer needed

Comment (romaxa) [reply examples/deck/example.js]: This option is used to enable "split" tool with 90 degree mode enabled, why are you suggesting to remove it?

File: docs/api-reference/mode-handlers/overview.md (hunk 1 above)
Comment (georgios-uber) [reply docs/api-reference/mode-handlers/overview.md]: @romaxa please update the doc like Clay suggested.

Comment (georgios-uber) [inline docs/api-reference/mode-handlers/overview.md]:
```suggestion
* `lock90Degree` (Boolean, optional)
  * Default: `false`

If true, all angles will be guaranteed to be 90 degrees.
```

**Pull request merged**